src/lib/containers/discussion_forum/DiscussionThread.tsx

```diff
@@ -35,21 +37,39 @@ export function DiscussionThread(props: DiscussionThreadProps) {
   const [showThreadModal, setShowThreadModal] = useState(false)
   const [showReplyEditor1, setShowReplyEditor1] = useState(false)
   const [showReplyEditor2, setShowReplyEditor2] = useState(false)
+  const [showDeleteModal, setShowDeleteModal] = useState(false)
 
   const { threadData, threadBody } = useGetThread(threadId)
   const { data: currentUserProfile } = useGetCurrentUserProfile()
+  const { data: moderatorList } = useGetModerators(threadData?.forumId ?? '', {
+    enabled: !!threadData,
+  })
   const { subscription, toggleSubscribed, isLoading } = useSubscription(
     threadId,
     SubscriptionObjectType.THREAD,
   )
-
+  const { mutate: deleteThread } = useDeleteThread({
+    onSuccess: () => {
+      setShowDeleteModal(false)
+      displayToast('A thread has been deleted.', 'info')
+    },
+  })
   const isAuthor = threadData?.createdBy == currentUserProfile?.ownerId
+  const isModerator = moderatorList?.results.includes(
+    currentUserProfile?.ownerId ?? '',
+  )
+
+  console.log(currentUserProfile)
 
   function handleFollowBtn() {
-    try {
-      toggleSubscribed()
-    } catch (err: any) {
-      displayToast(err.reason as string, 'danger')
+    if (currentUserProfile?.userName == 'anonymous') {
+      alert('You will need to sign in for access to that resource.')
```

Comment:
Prevents anonymous user to use functionality such as subscribing or posting a comment. Would there be a better way than using `alert`?

Reply:
There is the SRC_SIGN_IN_CLASS (see HasAccessV2 for complications in using this) if you want to attempt to automatically invoke the sign-in UI

Reply:
I would refactor to make the message a reusable constant.
There may be an opportunity here to introduce an abstraction for all of preflight checks (that a user must pass before doing certain operations). This could start as a signed-in check, but could be expanded to check for certification status or a validated user profile.  Just a thought.

Reply:
I would not use the native browser alert functionality, as it is not aligned in style.  I think using displayToast is preferable.